src/agora/txrx/workers/txrx_worker_hw.cc

```diff
@@ -448,15 +449,35 @@ size_t TxRxWorkerHw::DoTx(long long time0) {
       if (kDebugDownlink == true) {
         for (size_t ch = 0; ch < channels_per_interface_; ch++) {
           // Not exactly sure why 0 index was selected here.  Could it be a beacon ant?
-          if (ant_id != 0) {
+          if (!kBeamsweep && ant_id != 0) {
             txbuf.at(ch) = zeros_.data();
           } else if (dl_symbol_idx <
                      Configuration()->Frame().ClientDlPilotSymbols()) {
-            txbuf.at(ch) =
-                reinterpret_cast<void*>(Configuration()->UeSpecificPilotT()[0]);
+            std::complex<int16_t> pilot[Configuration()->SampsPerSymbol()];
```

Comment:
Might be reasonable to use a preallocated array (thread member) for scratch memory "pilot"